muswarmadmin/repositories.py

```diff
@@ -38,6 +48,40 @@
         await app.update_state(project_id, SwarmUI.Down)
 
 
+async def repository_has_location(app, pipeline):
+    """
+    Check if the repository associated with the pipeline
+    has a location (git url).
+    """
+    result = await app.sparql.query("""
+        ASK
+        FROM {{graph}}
+        WHERE {
+         ?repository ?p {{pipeline}} .
+         ?repository doap:location ?location .
+        }
+    """, pipeline=pipeline)
+    return result['boolean']
```

Comment:
I placed all the helpers in `main.py` because they could be used somewhere else. Obviously it isn't the case for this one. But I believe the module may get split to different modules if the number of lines in the file gets too big (500 is usually a good indicator in my experience).